### ISSUE TITLE
pyright does not suggest methods bound to class after its definition

### ISSUE THREAD

**Issue by orlandini** [microsoft/pyright#1621]
**Bug description**
I have a class defined in an extension module that is imported in a python module. In the python module, methods are added to the imported class. pyright does not show these methods in the completion suggestions (I am using pyright through `lsp-pyright` on emacs)

**To Reproduce**
The following example was enough to reproduce the issue for me:
```
class TestClass:
    def MyMethod(self):
        print("MyMethod")


TestClass.MyOtherMethod = lambda custom_str : print(custom_str)

tscl = TestClass
tscl.M
```

**Expected behavior**
I would expect pyright to suggest `MyMethod` and `MyOtherMethod`, however it only suggests `MyMethod`

**Screenshot**
![Screenshot from 2021-03-11 13-48-37](https://user-images.githubusercontent.com/11822222/110823335-c354e180-8270-11eb-9849-4808f1c0997d.png)

**VS Code extension or command-line**
I am running pyright through lsp-pyright on emacs.

I am not 100% sure that this issue belongs here, please let me know if that is not the case.

**Comment by erictraut** [comment]
This is intended behavior. What you're doing here is a type violation. If you enabled typeCheckingMode, you would see an error when you attempt to assign a value to `TestClass.MyOtherMethod` because that attribute is not defined for the class `TestClass`. Static type checkers don't allow for dynamic and ad-hoc addition of attributes to a class or an instance. You need to declare attributes within the class definition.

```python
class TestClass:
    def MyMethod(self):
        print("MyMethod")

    MyOtherMethod = lambda custom_str : print(custom_str)
```

or

```python
from typing import Callable

class TestClass:
    MyOtherMethod: Callable[[str], None]
    
    def MyMethod(self):
        print("MyMethod")

TestClass.MyOtherMethod = lambda custom_str : print(custom_str)
```

**Comment by orlandini** [comment]
Thanks a lot for your explanation, @erictraut.
